ferrischat_webserver/src/auth/reset_password.rs

```diff
@@ -0,0 +1,152 @@
+use crate::auth::generate_random_bits;
+use crate::WebServerError;
+use axum::extract::Json;
+use axum::extract::Path;
+use ferrischat_common::request_json::PasswordResetJson;
+use ferrischat_common::types::{ErrorJson, SuccessJson};
+use ferrischat_redis::{redis::AsyncCommands, REDIS_MANAGER};
+use lettre::{
+    transport::smtp::authentication::Credentials, AsyncSmtpTransport, AsyncTransport, Message,
+    Tokio1Executor,
+};
+
+/// POST /v0/auth/reset
+/// Requires an authtoken and a new password encoded in JSON, like: ```
+/// {
+/// "password": "ASecurePassword"
+/// }```
+pub async fn reset_password(
+    Json(PasswordResetJson { password }): Json<PasswordResetJson>,
+    crate::Authorization(authorized_user): crate::Authorization,
```

Comment:
Users that need to reset their password would not have access to their auth token.

Reply:
I built half of each system, it seems